lib/Mongo/MongoCommandCursor.php

```diff
@@ -112,4 +124,37 @@ public function __sleep()
     {
         return ['command'] + parent::__sleep();
     }
+
+    /**
+     * @see https://github.com/mongodb/mongo-php-driver-legacy/blob/1.6.14/db.c#L51
+     * @return bool
+     */
+    private function supportsReadPreference()
+    {
+        if ($this->command === []) {
+            return false;
```

Comment:
Is this merely guarding against the assumption below that there is at least one key in the command document?

Is the command document always a PHP array here, or might the user have supplied a stdClass? `ext-mongo` uses PHP's Hash API to access keys and would work with either.

Reply:
I took a look and I typehinted `array` in the constructor of `MongoCommandCursor`, although I might have to loosen that up depending on what `ext-mongo` does. For now, the check against an empty array works.